infor-design/enterprise
# 3336 - Add code to detect wkWebView [v4.25.x]

**Explain the _details_ for making this change. What existing problem does the pull request solve?**

This code adds detection to environment.js for wkWebView. It will treat wkWebview similar to safari by adding both classes to the head. Also added info on the about if needed and fixed all places in js (one in field options) where safari detection is used.

**Related github/jira issue (required)**:
Fixes #3336 
https://github.com/infor-design/enterprise/issues/3331 is related but has a different fix.

**Steps necessary to review your pull request (required)**:
- un zip the attachment which is a demo app for IOS simulator [SOBrowserApp-master.zip](https://github.com/infor-design/enterprise/files/4098007/SOBrowserApp-master.zip) so you can test wkWebView
- run this in xcode (latest version)
- press go in the app
- the page should show wkWebView
- type in http://localhost:4000/components/about/example-index.html and press go
- open the about (it should detect wkWebView)
- also go to http://localhost:4000/components/applicationmenu/example-index.html and open the menu -> the text should look normal not all black like the issue.
- test http://localhost:4000/components/environment/example-browser-specs.html in safari

**Included in this Pull Request**:x
